src/finalfusion/metadata.py

```diff
@@ -0,0 +1,76 @@
+"""
+finalfusion metadata
+"""
+from os import PathLike
+from typing import BinaryIO, Union
+
+import toml
+
+from finalfusion.io import Chunk, ChunkIdentifier, find_chunk, _read_binary, _write_binary
+
+
+class Metadata(dict, Chunk):
+    """
+    Embeddings metadata
+
+    Metadata can be used as a regular Python dict. For serialization, the contents need to be
+    serializable through `toml.dumps`. Finalfusion assumes metadata to be a TOML formatted
+    string.
+
+    Examples
+    --------
+    >>> metadata = Metadata({'Some': 'value', 'number': 1})
+    >>> metadata
+    {'Some': 'value', 'number': 1}
+    >>> metadata['Some']
+    'value'
+    >>> metadata['Some'] = 'other value'
+    >>> metadata['Some']
+    'other value'
+    """
+    @staticmethod
+    def chunk_identifier():
+        return ChunkIdentifier.Metadata
+
+    @staticmethod
+    def read_chunk(file: BinaryIO) -> 'Metadata':
+        file.seek(-12, 1)
```

Comment:
Replace magic offset. It is not immediate clear where `-12` comes from. This should be a descriptive constant or something else that makes it clear.